lib/src/components/NxSidebarNavigation/stateful/types.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import * as PropTypes from 'prop-types';
+import { WeakValidationMap } from 'react';
+
+import { SidebarNavLinkProps, sidebarNavLinkPropTypes } from '../types';
+
+export interface Props {
+  isDefaultOpen: boolean;
+  className?: string | null;
+  onToggleClick?: ((newToggleState: boolean) => void) | null;
+  logoImg: string;
+  logoText: string;
+  logoLink: string;
+  links: SidebarNavLinkProps[];
+  helpLink?: string | null;
+  helpText?: string | null;
+  collapsedReleaseText?: string | null;
+  expandedReleaseText?: string | null;
+  attributions?: string[] | null;
+}
+
+export const propTypes: WeakValidationMap<Props> = {
```

Comment:
Use `ValidationMap` if possible.  `WeakValidationMap` doesn't really work right in terms of enforcing that the TS props and react propTypes match